### DISCUSSION TITLE
[indexer] Get ready to nearcore 1.27.0

### DISCUSSION THREAD

**Discussion by khorolets** [near/nearcore#6884] [Node Public Interfaces]
# TL;DR `fa305730a0b1fe048c26ff528a0e188a3796df2e`

- `testnet` May 25, 2022
- `mainnet` June 22, 2022

*This release will make a protocol upgrade, your nodes might get stuck if you don't upgrade them before the protocol upgrade happens.*

---

## NEAR Lake Framework

We want to remind you that we've [released NEAR Lake Framework](https://gov.near.org/t/announcement-near-lake-framework-brand-new-word-in-indexer-building-approach/17668). If you don't have a hard requirement to use NEAR Indexer Framework and run `nearcore` node for your indexer, we encourage you to migrate to [NEAR Lake Framework](https://github.com/near/near-lake-framework).

NEAR Lake Framework allows you to build a lightweight indexer which doesn't require a lot of resources. This will:
- decrease your spending on infrastructure
- don't require updates on every `nearcore` release

Please, reach out to us if you have questions about migration, we will be happy to help.

---

## Commit hash

[`fa305730a0b1fe048c26ff528a0e188a3796df2e`](https://github.com/near/nearcore/commit/fa305730a0b1fe048c26ff528a0e188a3796df2e)

## Expected issues

No issues were detected. You may encounter some dependency versions conflicts, but `Cargo.lock` from `nearcore` trick should solve them immediately

> ### `Cargo.lock` trick
> Replace the `Cargo.lock` file in your project with the [`Cargo.lock`](https://raw.githubusercontent.com/near/nearcore/fa305730a0b1fe048c26ff528a0e188a3796df2e/Cargo.lock) file from `nearcore` release and run `cargo check` or `cargo build`

## Reference

See how we did the upgrade on the [NEAR Indexer for Explorer side](https://github.com/near/near-indexer-for-explorer/pull/275/files)

---

Feel free to leave your questions in the comments below.
